runtime-sdk/src/modules/consensus/mod.rs

```diff
@@ -0,0 +1,278 @@
+//! Consensus module.
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use oasis_core_runtime::consensus::{
+    roothash::{Message, StakingMessage},
+    staking,
+    staking::Account as ConsensusAccount,
+    state::{staking::ImmutableState as StakingImmutableState, StateError},
+};
+
+use crate::{
+    context::{Context, DispatchContext},
+    module,
+    module::Module as _,
+    modules,
+    types::{address::Address, message::MessageEventHookInvocation, token},
+};
+
+#[cfg(test)]
+mod test;
+
+/// Unique module name.
+const MODULE_NAME: &str = "consensus";
+
+/// Parameters for the consensus module.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Parameters {
+    pub consensus_denomination: token::Denomination,
+}
+
+impl Default for Parameters {
+    fn default() -> Self {
+        Self {
+            consensus_denomination: token::Denomination::from_str("TEST").unwrap(),
+        }
+    }
+}
+
+impl module::Parameters for Parameters {
+    type Error = ();
+}
+/// Events emitted by the consensus module (none so far).
+#[derive(Debug, Serialize, Deserialize, oasis_runtime_sdk_macros::Event)]
+#[serde(untagged)]
+pub enum Event {}
+
+/// Genesis state for the consensus module.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Genesis {
+    #[serde(rename = "parameters")]
+    pub parameters: Parameters,
+}
+
+#[derive(Error, Debug, oasis_runtime_sdk_macros::Error)]
+pub enum Error {
+    #[error("invalid argument")]
+    #[sdk_error(code = 1)]
+    InvalidArgument,
+
+    #[error("invalid denomination")]
+    #[sdk_error(code = 2)]
+    InvalidDenomination,
+
+    #[error("core error")]
+    #[sdk_error(code = 3)]
```

Comment:
```suggestion
    #[sdk_error(transparent)]
```
(After rebase on master.)